source_manager.go

```diff
@@ -27,6 +27,18 @@ var sanitizer = strings.NewReplacer(":", "-", "/", "-", "+", "-")
 // sufficient for any purpose. It provides some additional semantics around the
 // methods defined here.
 type SourceManager interface {
+	// SourceManager adds a Release method to the sourceDeducer method set. The
+	// sourceDeducer method set is defined in a separate interface because it's
+	// used by the sourceBridge interface as well, but sourceBridges don't get a
+	// Release method.
+	sourceDeducer
```

Comment:
Yep, to all of the above. (Of those four, I hate `bridge` the most - it's not even the right pattern name. Though I think `Deducer` is OK).